lib/ClangImporter/ClangImporter.cpp

```diff
@@ -6218,6 +6222,21 @@ TinyPtrVector<ValueDecl *> ClangRecordMemberLookup::evaluate(
           // as that would cause an ambiguous lookup.
           if (foundNameArities.count(getArity(foundInBase)))
             continue;
+
+          // Do not importBaseMemberDecl() if this is a recursive lookup into
+          // some class's superclass. importBaseMemberDecl() caches synthesized
+          // members, which does not work if we call it on its result, e.g.:
+          //
+          //    importBaseMemberDecl(importBaseMemberDecl(foundInBase,
```

Comment:
Could we still run into a similar problem if the same `recordDecl` is accessible on multiple paths in the inheritance DAG? E.g.:
```
struct A {};
struct B : A {};
struct C : B, A {};
```